pom.xml

```diff
@@ -39,6 +39,27 @@
       <scope>provided</scope>
       <optional>true</optional>
     </dependency>
+    <!-- include junit-platform-runner to run JUnit 5 tests with JUnit 4,
+      can be removed after upgrading to maven-surefire-plugin v2.22.0, see SUREFIRE-1330 -->
+    <dependency>
+      <groupId>org.junit.platform</groupId>
+      <artifactId>junit-platform-runner</artifactId>
+      <version>1.2.0</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>5.2.0</version>
+      <scope>provided</scope>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>5.2.0</version>
+      <scope>test</scope>
```

Comment:
I believe this should be option and provided as in `junit-jupiter-api`

Reply:
Yes, other projects should also have junit-jupiter-engine as dependency. But the only dependency, projects should depend on, is junit-jupiter-api. The junit-jupiter-engine is only necessary on test-runtime.

An alternative would be to declare junit-jupiter-engine as dependency of maven-surefire-plugin.
```
<plugin>
	<artifactId>maven-surefire-plugin</artifactId>
	<dependencies>
		<dependency>
			<groupId>org.junit.jupiter</groupId>
			<artifactId>junit-jupiter-engine</artifactId>
		</dependency>
	</dependencies>
</plugin>
```

See [Running Tests with Maven](https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven)

Reply:
I don't know much about plugin dependencies, what we want here is not to drag junit 5 dependencies for users that don't use it yet. would your suggestion achieve that goal ?

Reply:
For users of assertj-core the current solution would be fine. They already have a dependency to junit-jupiter-api in test-compile scope and junit-jupiter-engine in test-runtime scope.

For those who want to build the code of assertj-core both solutions are fine, either junit-jupiter-engine as test-compile dependency (`<scope>test</scope>`) or as test-runtime dependency (the suggested dependency of maven-surefire-plugin). But I think there was an issue when running tests within an older IDE that doesn't already support JUnit 5, I think they need junit-jupiter-engine in test scope.

So I would keep the current solution with a test-compile dependency.